integration_tests/serviceprovider_mock.go

```diff
@@ -36,6 +36,7 @@ type TestServiceProvider struct {
 	CheckRepositoryAccessImpl func(context.Context, client.Client, *api.SPIAccessCheck) (*api.SPIAccessCheckStatus, error)
 	MapTokenImpl              func(context.Context, *api.SPIAccessTokenBinding, *api.SPIAccessToken, *api.Token) (serviceprovider.AccessTokenMapper, error)
 	ValidateImpl              func(context.Context, serviceprovider.Validated) (serviceprovider.ValidationResult, error)
+	CustomizeReset            func()
```

Comment:
Just from the purity point of view, I would make this accept the instance of the `TestServiceProvider` that is being customized so that the impl doesn't have to assume that the instance is available through a global variable.

I.e.
```go
CustomizeReset func(*TestServiceProvider)
```